fair_drop/suspicious.py

```diff
@@ -0,0 +1,205 @@
+    respect_retry_after_header=True,
+)
+adapter = HTTPAdapter(max_retries=retry_strategy)
+scraper.mount("https://", adapter)
+scraper.mount("http://", adapter)
+
+
+def is_nft_suspicious(nft_url):
+    logging.debug(f"Scraping NFT with link: {nft_url}")
+    res = scraper.get(nft_url)
+
+    if res.status_code == 404:  # NFT not found
+        logging.info("NFT not found. Probably reached the end of a collection")
+        return None, None
+    if res.status_code == 200:
+        soup = BeautifulSoup(res.text, "html.parser")
+        collection_name = soup.find(class_="item--collection-detail").text
+        owner = soup.find(class_="AccountLink--ellipsis-overflow").text.replace(
+            "Owned by\xa0", ""
+        )
+        data = {
+            "collection": args.collection_address,
+            "collection_name": collection_name,
+            "blockchain": "ethereum",
+            "url": nft_url,
+            "owner": owner,
+        }
+        if res.text.find("Reported for suspicious") > 0:
+            logging.info(f"Found suspicious NFT of URL {nft_url}")
+            return True, data
+        else:
+            return False, data
+
+
+OPENSEA_BASE_URL = (
+    "https://opensea.io/assets/ethereum/"  # TODO adjust for other blockchains
+)
+
+
+def list_collection_nfts_urls(collection_address):
+    """List all OpenSea URLs of NFTs in a collection
+
+    Args:
+        collection_address (string): NFT collection address
+
+    Returns:
+        array: list of the OpenSea URLs of NFTs
+    """
+    # ! This is just a mock function. It is to be replaced with a call to the OpenSea API
+    nft_urls = []
+    for i in range(lower_id, upper_id + 1):
+        nft_urls.append(f"{OPENSEA_BASE_URL}{collection_address}/{i}")
+    return nft_urls
+
+
+def scrape_all_collection_suspicious_nfts(collection_address):
+    # TODO check that the collection address is valid
+    collection_nfts_urls = list_collection_nfts_urls(collection_address)
+    logging.info(f"Collection contains {len(collection_nfts_urls)} NFTs")
+    collection_cache = load_scrape_cache(collection_nfts_urls)
+    logging.info(f"Found {len(collection_cache)} NFTs in collection cache")
+    # TODO removed scraped URLs from the list to be scraped
+    for index, nft in collection_cache.iterrows():
+        if (
+            nft["url"] in collection_nfts_urls
+        ):  # TODO add an expiry rule, depending on date of last scraped
+            logging.debug(f"NFT to be scraped already in cache. Skipping {nft['url']}")
+            collection_nfts_urls.remove(nft["url"])
+    logging.info(f"Scraping a list of {len(collection_nfts_urls)} NFTs")
+    BATCH_SIZE = int(args.batch_size)
+    nft_urls_batches = [
+        collection_nfts_urls[i : i + BATCH_SIZE]
+        for i in range(0, len(collection_nfts_urls), BATCH_SIZE)
+    ]
+    for index, batch in enumerate(nft_urls_batches):
+        print(f"Scraped {index * BATCH_SIZE} NFT URLs so far")
+        with Pool(5) as p:
```

Comment:
```suggestion
        with multiprocessing.Pool(multiprocessing.cpu_count() - 1) as p:
```
This approach should considerably shorten the execution time for users with multiple cores. 
In my case it reduced the time by 35%.